docs/cookbook/extending-product-list.md

```diff
@@ -66,9 +66,116 @@ class ListedProductView extends BaseListedProductView
+     * @param string $locale
+     * @param int $startFrom
+     * @param int $batchSize
+     * @return array
+     */
+    public function getProductsData(int $domainId, string $locale, int $startFrom, int $batchSize): array
+    {
+        $queryBuilder = $this->createQueryBuilder($domainId, $locale)
+            ->setFirstResult($startFrom)
+            ->setMaxResults($batchSize);
+
+        $query = $queryBuilder->getQuery();
+
+        $result = [];
+        /** @var \Shopsys\FrameworkBundle\Model\Product\Product $product */
+        foreach ($query->getResult() as $product) {
+            $flagIds = $this->extractFlags($product);
+            $categoryIds = $this->extractCategories($domainId, $product);
+            $parameters = $this->extractParameters($locale, $product);
+            $prices = $this->extractPrices($domainId, $product);
+            $visibility = $this->extractVisibility($domainId, $product);
+
+            $friendlyUrl = $this->friendlyUrlRepository->getMainFriendlyUrl($domainId, 'front_product_detail', $product->getId());
+
+            $result[] = [
+                'id' => $product->getId(),
+                'catnum' => $product->getCatnum(),
+                'partno' => $product->getPartno(),
+                'ean' => $product->getEan(),
+                'name' => $product->getName($locale),
+                'description' => $product->getDescription($domainId),
+                'shortDescription' => $product->getShortDescription($domainId),
+                'brand' => $product->getBrand() ? $product->getBrand()->getId() : '',
+                'brand_name' => $product->getBrand() ? $product->getBrand()->getName() : '',
+                'flags' => $flagIds,
+                'categories' => $categoryIds,
+                'in_stock' => $product->getCalculatedAvailability()->getDispatchTime() === 0,
+                'prices' => $prices,
+                'parameters' => $parameters,
+                'ordering_priority' => $product->getOrderingPriority(),
+                'calculated_selling_denied' => $product->getCalculatedSellingDenied(),
+                'selling_denied' => $product->isSellingDenied(),
+                'availability' => $product->getAvailability()->getName($locale),
+                'main_variant' => $product->isMainVariant(),
+                'detail_url' => $friendlyUrl->getAbsoluteUrl($this->domain),
+                'visibility' => $visibility,
+            ];
+        }
+
+        return $result;
+    }
+}
+```
+
+You need to register your new class as an alias for the one from the FrameworkBundle in `services.yml`:
+
+```yml
+Shopsys\FrameworkBundle\Model\Product\Search\Export\ProductSearchExportWithFilterRepository: '@Shopsys\ShopBundle\Model\Product\Search\Export\ProductSearchExportWithFilterRepository'
+```
+
+Then you need to fix `ProductSearchExportRepositoryTest::getExpectedStructureForRepository` (because this test check if our structure is correct) by adding new attribute:
```

Comment:
```suggestion
Then you need to fix `ProductSearchExportRepositoryTest::getExpectedStructureForRepository` (because this test check if your structure is correct) by adding new attribute:
```